nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/test/java/org/apache/nifi/processors/solr/TestFetchSolr.java

```diff
@@ -0,0 +1,380 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.xmlunit.matchers.CompareMatcher;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+
+public class TestFetchSolr {
+    static final String DEFAULT_SOLR_CORE = "testCollection";
+
+    private static final SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US);
+    static {
+        df.setTimeZone(TimeZone.getTimeZone("GMT"));
+    }
+
+    private SolrClient solrClient;
+
+    @Before
+    public void setup() {
+
+        try {
+
+            // create an EmbeddedSolrServer for the processor to use
+            String relPath = getClass().getProtectionDomain().getCodeSource()
+                    .getLocation().getFile() + "../../target";
+
+            solrClient = EmbeddedSolrServerFactory.create(EmbeddedSolrServerFactory.DEFAULT_SOLR_HOME,
+                    DEFAULT_SOLR_CORE, relPath);
+
+            for (int i = 0; i < 10; i++) {
+                SolrInputDocument doc = new SolrInputDocument();
+                doc.addField("id", "doc" + i);
+                Date date = new Date();
+                doc.addField("created", df.format(date));
+                doc.addField("string_single", "single" + i + ".1");
+                doc.addField("string_multi", "multi" + i + ".1");
+                doc.addField("string_multi", "multi" + i + ".2");
+                doc.addField("integer_single", i);
+                doc.addField("integer_multi", 1);
+                doc.addField("integer_multi", 2);
+                doc.addField("integer_multi", 3);
+                doc.addField("double_single", 0.5 + i);
+
+                solrClient.add(doc);
+                System.out.println(doc.getField("created").getValue());
+
+            }
+            solrClient.commit();
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    @After
+    public void teardown() {
+        try {
+            solrClient.close();
+        } catch (Exception e) {
+        }
+    }
+
+    @Test
+    public void testAllFacetCategories() throws IOException {
+        final TestableProcessor proc = new TestableProcessor(solrClient);
+
+        TestRunner runner = TestRunners.newTestRunner(proc);
+        runner.setProperty(SolrUtils.SOLR_TYPE, SolrUtils.SOLR_TYPE_CLOUD.getValue());
+        runner.setProperty(SolrUtils.SOLR_LOCATION, "http://localhost:8443/solr");
+        runner.setProperty(SolrUtils.COLLECTION, "testCollection");
+        runner.setProperty(FetchSolr.SOLR_QUERY_STRING, "q=*:*" +

```

Comment:
Again, might want to think about multiple query-related fields so that this can be spread out and tested piece by piece by the user.